app/containers/RoomHeader/index.tsx

```diff
@@ -1,124 +1,79 @@
+	visitor?: IVisitor;
 }
 
-class RoomHeaderContainer extends Component<IRoomHeaderContainerProps, any> {
-	shouldComponentUpdate(nextProps: IRoomHeaderContainerProps) {
-		const {
-			type,
-			title,
-			subtitle,
-			status,
-			statusText,
-			connecting,
-			connected,
-			onPress,
-			usersTyping,
-			width,
-			height,
-			teamMain,
-			sourceType
-		} = this.props;
-		if (nextProps.type !== type) {
-			return true;
-		}
-		if (nextProps.title !== title) {
-			return true;
-		}
-		if (nextProps.subtitle !== subtitle) {
-			return true;
-		}
-		if (nextProps.status !== status) {
-			return true;
-		}
-		if (nextProps.statusText !== statusText) {
-			return true;
-		}
-		if (nextProps.connecting !== connecting) {
-			return true;
-		}
-		if (nextProps.connected !== connected) {
-			return true;
-		}
-		if (nextProps.width !== width) {
-			return true;
-		}
-		if (nextProps.height !== height) {
-			return true;
-		}
-		if (!dequal(nextProps.usersTyping, usersTyping)) {
-			return true;
-		}
-		if (!dequal(nextProps.sourceType, sourceType)) {
-			return true;
-		}
-		if (nextProps.onPress !== onPress) {
-			return true;
-		}
-		if (nextProps.teamMain !== teamMain) {
-			return true;
-		}
-		return false;
-	}
+const RoomHeaderContainer = React.memo(
+	({
+		isGroupChat,
+		onPress,
+		parentTitle,
+		prid,
+		roomUserId,
+		subtitle: subtitleProp,
+		teamMain,
+		testID,
+		title,
+		tmid,
+		type,
+		sourceType,
+		visitor
+	}: IRoomHeaderContainerProps) => {
+		const [subtitle, setSubtitle] = useState<string | undefined>(undefined);
+		const [status, setStatus] = useState<TUserStatus>('offline');
+		const [statusText, setStatusText] = useState<string | undefined>(undefined);
+		const { width, height } = useDimensions();
 
-	render() {
-		const {
-			title,
-			subtitle: subtitleProp,
-			type,
-			teamMain,
-			prid,
-			tmid,
-			status = 'offline',
-			statusText,
-			connecting,
-			connected,
-			usersTyping,
-			onPress,
-			width,
-			height,
-			parentTitle,
-			isGroupChat,
-			testID,
-			sourceType
-		} = this.props;
+		const connecting = useSelector((state: IApplicationState) => state.meteor.connecting || state.server.loading);
+		const usersTyping = useSelector((state: IApplicationState) => state.usersTyping, shallowEqual);
+		const connected = useSelector((state: IApplicationState) => state.meteor.connected);
+		const activeUser = useSelector(
+			(state: IApplicationState) => (roomUserId ? state.activeUsers?.[roomUserId] : undefined),
+			shallowEqual
+		);
 
-		let subtitle;
-		if (connecting) {
-			subtitle = I18n.t('Connecting');
-		} else if (!connected) {
-			subtitle = I18n.t('Waiting_for_network');
-		} else {
-			subtitle = subtitleProp;
-		}
+		useEffect(() => {
+			if (connecting) {
+				setSubtitle(I18n.t('Connecting'));
+			} else if (!connected) {
+				setSubtitle(I18n.t('Waiting_for_network'));
+			} else {
+				setSubtitle(subtitleProp);
+			}
+		}, [connecting, connected]);
+
+		useEffect(() => {
```

Comment:
You don't have to create a state to render the value you want.
You already receive those values from `props`, the same way we did on `mapStateToProps`.
If any of those `useSelector` changes, the re-render is going to be triggered and your value would work just fine.
Adding this extra state the way you did is just re-rendering unnecessarily.